app/Console/Commands/LdapAdSync.php

```diff
@@ -0,0 +1,380 @@
+<?php
+
+declare(strict_types=1);
```

Comment:
What's the benefit of adding this here? I generally don't like having things that would otherwise seem to be global settings set on an ad hoc basis. We do it for timeouts within some of the LDAP methods just because some OU's can be very large, but is this actually required?

Reply:
This is a good explanation https://stackoverflow.com/questions/48723637/what-does-strict-types-do-in-php

But basically it makes for better code

Reply:
I know what strict typing is/does, I'm just not sure I want that declared at the top of individual files. This can lead to inconsistent behavior throughout the system, since the entire system isn't using the same settings. If we're going to try to go strict typing, we should do to everywhere, not in specific files, IMHO. 

(I'm also not generally convinced strict typing is even a good idea in PHP, to be honest. I've found it to create as many problems as it prevents.) 

Overall this PR looks really good - I need to figure out how to test it though. I have access to LDAP, but not AD.

Reply:
I've use strict types all time, even with old code bases and never had a problem. I can remove them if you want. 
I found this, https://documize.github.io/ad-ldap-test-server/ to test AD.

Reply:
>  I found this, https://documize.github.io/ad-ldap-test-server/ to test AD.

Oooh, will take a look at that. We use JumpCloud for LDAP testing, and there used to be an AD server that someone made available for free for testing (would reset every hour or whatever), but it's been down for years, and they never responded. :-/